precompiles/staking/method_distribute.go

```diff
@@ -0,0 +1,116 @@
+package staking
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	ptypes "github.com/zeta-chain/node/precompiles/types"
+	fungibletypes "github.com/zeta-chain/node/x/fungible/types"
+)
+
+var (
+	zrc20lockerAddress = common.HexToAddress("0x0000000000000000000000000000000000000067")
+)
+
+// function distribute(address zrc20, uint256 amount) external returns (bool success)
+func (c *Contract) distribute(
+	ctx sdk.Context,
+	evm *vm.EVM,
+	contract *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != 2 {
+		return nil, &(ptypes.ErrInvalidNumberOfArgs{
```

Comment:
curious why not `&ptypes.ErrInvalidNumberOfArgs{...}`

Reply:
Probably a typo, the result is the same though. Modified in https://github.com/zeta-chain/node/commit/06d0c5d5a8a2d7ce54f84a666c7bef7784963ccf.